src/network/networkd-network.h

```diff
@@ -135,6 +135,7 @@ struct Network {
         int dhcp_use_gateway;
         bool dhcp_use_timezone;
         bool dhcp_use_hostname;
+        bool dhcp_set_static_hostname;
```

Comment:
Maybe, its better to introduce a enum:
```
typedef enum UseHostname {
    USE_HOSTNAME_NO,
    USE_HOSTNAME_YES,
    USE_HOSTNAME_STATIC,
    _USE_HOSTNAME_MAX,
    _USE_HOSTNAM_INVALID = -1,
} UseHostname;
```
and then introduce a `_from_string()` with `DEFINE_PRIVATE_STRING_TABLE_LOOKUP_FROM_STRING_WITH_BOOLEAN()`.